lib/connection/client.rb

```diff
@@ -1,18 +1,87 @@
 module Connection
   class Client
     include Connection
-    include Connection::IOMethods
 
-    attr_reader :host
-    attr_reader :port
+    attr_reader :io
 
-    def initialize(host, port)
-      @host = host
-      @port = port
+    def initialize(io)
+      @io = io
     end
 
-    def socket
-      @socket ||= TCPSocket.new host, port
+    def self.build(io, scheduler=nil)
+      instance = new io
+      instance.configure_dependencies scheduler: scheduler
+      instance
+    end
+
+    def close
+      io.close
+      telemetry.closed
+
+    rescue IOError => error
+      telemetry.closed
+      raise error
+    end
+
+    def readline(*arguments)
+      readline_command.(*arguments)
+    end
+    alias_method :gets, :readline
+
+    def max_read_size
+      8192
```

Comment:
Could this magic number be problematic with more generalized use of this library?